pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -258,6 +258,15 @@ in
     }
   );
 
+  aw-watcher-vim = super.aw-watcher-vim.overrideAttrs {
+    patches = [
+      (substituteAll {
+        src = ./patches/aw-watcher-vim/program_paths.patch;
+        curl = lib.getExe curl;
```

Comment:
Surprised that this, instead of
```nix
"@curl@" = lib.getExe curl;
```
works

Reply:
@PerchunPak I mean, I am not an expert but I looked through `vim/plugins/overrides.nix` and other uses of `substituteAll` + `*.patch` files and it's like this everywhere, quite on the contrary even, I can't find any examples of the `"@pkg@" = pkg;` pattern that you mentioned ... and any larger changes are a bit out of the scope of this request.

Reply:
Just skimmed through https://github.com/NixOS/nixpkgs/blob/12f623110b58ef485ab4c8e140190586bd5e9229/pkgs/stdenv/generic/setup.sh#L1025 and it seems like its being appended if you don't specify it in the attribute name? Possibly? Perhaps? Mayhaps? :thinking:

Reply:
Yes, you are not supposed to add the `@` indeed.

Reply:
Ignore my comment, it's not a suggestion, it's just a note. Perhaps I badly worded it